server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/utils/ErrorResponder.java

```diff
@@ -89,12 +96,28 @@ public HaltException haltError() {
             return halt(statusCode, new JsonTransformer().render(new ErrorDetail(statusCode,
                 type.getType(),
                 message,
-                cause.map(e -> Optional.ofNullable(e.getMessage())).orElse(Optional.empty()))));
+                cause.map(Throwable::getMessage))));
         } catch (JsonProcessingException e) {
             return halt(statusCode);
         }
     }
 
+    public String toResponse(Response response) {
+        response.status(statusCode);
+        try {
+            String body = new JsonTransformer().render(new ErrorDetail(
```

Comment:
There is redoundancy with haltError method.

Maybe we should extract Json serialization inside a dedicated method?